Sources/ArcGISToolkit/Extensions/ArcGIS/AttachmentHelpers/FeatureAttachment.swift

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Foundation
+import UIKit
+
+/// The type of attachments in a FeatureAttachment.
+public enum FeatureAttachmentKind {
+    /// An image.
+    case image
+    /// A video.
+    case video
+    /// A document.
+    case document
+    /// An attachment of another type.
+    case other
+}
+
+public protocol FeatureAttachment: Loadable {    
+    /// The underlying ``Attachment``.
+    var attachment: Attachment? { get }
+    
+    /// The MIME content type of the attachment.
+    var contentType: String { get }
+    
+    /// The type of the attachment.
+    var featureAttachmentKind: FeatureAttachmentKind { get }
+
+    /// The local temporary filepath where we store the attachment once it is loaded.
+    var fileURL: URL? { get }
+    
+    /// A value indicating whether "loading" (fetching the data) can be accomplished without using the network.
+    ///
+    /// This is `true` if it just needs to pull the data from a database,
+    /// `false` if the loading will cause a network request.
+    var isLocal: Bool { get }
+    
+    /// The name of the attachment.
+    var name: String { get }
+    
+    /// The size of the attachment in bytes.
+    var size: Int { get }
+    
+    // MARK: Methods
+    //
+    /// Creates asynchronously the full image for displaying the attachment in full screen or some UI larger than a thumbnail.
+    ///
+    /// This is only supported if the ``kind-swift.property`` is ``Kind-swift.enum/image``.
```

Comment:
This also creates a warning.

<img width="1053" alt="Screenshot 2024-04-24 at 4 20 45 PM" src="https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/16397058/42e71208-7e00-4f44-bf15-922a225f202f">

Did you possibly mean this?
```suggestion
    /// This is only supported if ``FeatureAttachment/featureAttachmentKind`` is ``FeatureAttachmentKind/image``.
```